### PR TITLE
Package linting

### PR DESCRIPTION
**Pull request by joshblum**
Depends on #6 (once that is merged I'll rebase this so it cleanly reflects only the lint changes). @kmax12 there are a lot of changes to make to make the project pass `flake8`.

```
$ flake8 featuretools | wc -l
     854
```

I cleaned the file `featuretools/computational_backends/calculate_feature_matrix.py` as an example. Let me know if you want me to proceed with adding linting to the build (and getting the other files to pass). If you have any preference for the flake8 config let me know before I continue.

### CODE REVIEW

**Comment by kmax12** [comment]
Thanks @joshblum. For the linting, let's turn off the 80 character limit check. For now, we will leave it to the developer to decide if going over 80 characters will increase the readability of the code. 

For example, in your changes to `calculate_feature_matrix.py`, I think the 80 chars on the docstring is great for readability, but most of the other changes don't  increase the readability of the code.

File: featuretools/__init__.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-import variable_types
-from .entityset.api import *
-from . import primitives
-from .synthesis.api import *
-from .primitives import Feature
-from .computational_backends.api import *
-from . import tests
-from .utils.pickle_utils import *
-import featuretools.demo
+import config  # noqa
```
Comment (kmax12) [inline featuretools/__init__.py]: I believe you can put `# flake8: noqa` at the top of the file rather than repeat on every line.

Comment (joshblum) [reply featuretools/__init__.py]: sounds good!

**Comment by joshblum** [comment]
@kmax12 @rwedge PR should be good to go. Let me know if you would like any changes! All builds will run [flake8](https://github.com/PyCQA/flake8) and [isort](https://github.com/timothycrosley/isort) to check for any syntax/PEP8 errors/import order errors. I recommend adding [autopep8](https://github.com/hhatto/autopep8) and an [isort plugin](https://github.com/timothycrosley/isort/wiki/isort-Plugins) to your editor to automatically apply these rules during development.

**Pull request merged**